test/e2e/e2e.go

```diff
@@ -285,6 +296,30 @@ func CreateServiceAccountAndBinding(clients *test.Clients, name string, logf log
 	return nil
 }
 
+// CreatePodAndServiceReady will create a Pod and Service, and wait for them to become ready
+func CreatePodAndServiceReady(clients *test.Clients, pod *corev1.Pod, routeName string, ns string, selector map[string]string, logf logging.FormatLogger, cleaner *test.Cleaner) (*corev1.Pod, error) {
```

Comment:
Rather than `routeName, ns, selector`, can we pass in a `*corev1.Service`?

Reply:
Yeah this would look clearer. Changed.